lib/phabricator/conduit_client.rb

```diff
@@ -81,11 +81,20 @@ def get_credentials_from_file
 
     def post(method, data, opts={})
       log.debug("Making a `#{method}` request with data: #{data.inspect}.")
-      response = JSON.parse(RestClient.post("#{@host}/api/#{method}", {
-        params: data.to_json,
-        output: 'json',
-        max_redirects: 0
-      }.merge(opts)))
+      response = RestClient.post("#{@host}/api/#{method}", {
+                    params: data.to_json,
+                    output: 'json',
+                    max_redirects: 0
+                  }.merge(opts)) do |res|
+                    case res.code
+                    when 300, 301, 302, 303, 304, 307, 308
+                      log.error("Conduit response: #{res.inspect}, response code: #{res.code}")
+                      raise Errors::ClientError.new(res.code), 'Conduit connection error: the host used in the url is used to redirect the request'
```

Comment:
According to the docs the `max_redirects` count should raise `RestClient::MaxRedirectsReached` in v 1.6.7 -the one we are using now- however, it seems like it didn't get raised for some reason, I checked their code and they were raising it if the status code is 303 only https://github.com/rest-client/rest-client/blob/v1.6.7/lib/restclient/abstract_response.rb#L43 , so I guess this change will do the trick :+1: however we can remove 304 since it is not relevant to the redirects